book/jvm-build-service.md

```diff
@@ -32,6 +32,10 @@ Once the `ArtifactBuild` object has been created JBS will then attempt to try an
 
 Once the source code location has been discovered a [`DependencyBuild`](#DependencyBuild) object is created. There are generally less `DependencyBuild` objects than there are `ArtifactBuild`, as multiple artifacts can come from the same build. The controller will then try and build the artifact, first it will run a build discovery pipeline, that attempts to determine possible ways of building the artifact. Once discovery is complete it uses this information to attempt to build the artifact in a trial and error manner.
 
+Note that the dependency discovery pipeline can check both configured shared repositories and the main repository
+(which may be e.g. a Quay.io repository) for a pre-existing build ; if found it will shortcut the pipeline avoiding the
+build and refer to those RebuiltArtifacts instead.
```

Comment:
```suggestion
The dependency discovery pipeline can check both configured shared repositories and the main repository
(for example a Quay.io repository) for a pre-existing builds. If a previous build is found, the pipeline will shortcut to avoiding
building the artifact and will instead refer to the found `RebuiltArtifact`s instead.
```

Reply:
We introduce the concept of `RebuiltArtifact` later after its first use.

Reply:
I have rephrased it slightly to try avoid the reference.